source/device_defender.c

```diff
@@ -477,7 +477,7 @@ static int s_get_metric_report_json(
 
             switch (custom_metrics_data[metric_index].metric->type) {
                 case DD_METRIC_NUMBER:
-                    cJSON_AddNumberToObject(item, "number", (double)custom_metrics_data[metric_index].data.number);
+                    cJSON_AddNumberToObject(item, "number", custom_metrics_data[metric_index].data.number);
```

Comment:
I'm not one to complain about function size very often, but good grief, this is... not great (not your fault obviously, I'm just grumbling at having to try and look over the thing as a whole).

Reply:
Yeah, it's rather large. In the future (maybe with the JSON changes?) I'll take a look at splitting this function down into multiple smaller functions so it is easier to read, debug, and work with.